Bowling.java

```diff
@@ -2,25 +2,91 @@
 
 public class Bowling {
 
-  private int totalPoints;
+  private Scoreboard scoreboard;
 
   public Bowling() {
-    totalPoints = 0; 
+    scoreboard = new Scoreboard();
+    scoreboard.addFrame();
   }
 
-  public Bowling bowl(int pinsDown) {
-		if (pinsDown < 0) {
-			throw new IllegalArgumentException("Cannot knock down negative pins. Bowl again.");
-		}
-    totalPoints += pinsDown;
+  public Bowling bowl(Integer pinsDown) {
+    if (pinsDown < 0) {
+      throw new IllegalArgumentException("Cannot knock down negative pins. Bowl again.");
+    }
+
+    Frame currentFrame = scoreboard.getCurrentFrame();
```

Comment:
Is `isFull()` poor?...it depends.  Is `Scoreboard` inappropriate?...it depends. :)  I'll take these on in reverse order...

I think I wouldn't have brought in a `Scoreboard` yet.  Having it, however, makes me think that `Bowling` is more of our [port/adapter](https://www.infoq.com/news/2014/10/exploring-hexagonal-architecture) (though that might not be the best read) to the outside world - converting from simple inputs like a number to domain objects and vice versa.  So `bowl(pinsDown)` should very quickly pass the message onto `Scoreboard`, just like `score()` does (in reverse since it's getting).

`Frame.isFull()` seems spot on to me since `Frame` is probably going to be best aware of that since it will have the two bowls in it.  Now that we're thinking of `Bowling` as the translation into the domain objects it should only know about a `Frame` if it needs to convert `pinsDown` into one.  I can totally see `Scoreboard` being given `pinsDown`, figuring out if a `Frame` needs to be created, then using that frame to figure out if the score should be updated.

In a different way of thinking, I could see where `Bowling` created a `Frame` with `pinsDown`, if `isFull()` was false, it waited for it to be full (by a second or maybe third call if we're worrying about the tenth frame), then it pass the `Frame` onto the `Scoreboard` to figure things out.

Hopefully, that clarifies some of my thinking around this?